portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,36 +19,55 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+
 import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import java.util.ArrayList;
 import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
 import com.andrewrs.sps.data.ListRecord;
-import java.util.ArrayList;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
     private ArrayList<ListRecord> records;
-  public void init()
-  {
-    records = new ArrayList<ListRecord>();    
-  }
+  public void init(){}
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException 
   {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    records = new ArrayList<ListRecord>(); 
     response.setContentType("text/json;");
+    Query query = new Query("message_log").addSort("timeStamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable()) 
+    {
+        records.add(new ListRecord((long)entity.getProperty("timeStamp"),(String)entity.getProperty("message") ));
+    }
 
     Gson gson = new Gson();
-    //
     response.getWriter().println(gson.toJson(records));
   }
     @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException 
   {
     String text = getParameter(request, "message", "");
     String messages[] = text.split(",");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+ 
     for(String message: messages)
     {
-      records.add(new ListRecord(LocalDateTime.now(),message));
+      Entity entity = new Entity("message_log");
+      entity.setProperty("timeStamp",System.currentTimeMillis());
+      entity.setProperty("message",message);
```

Comment:
Nit: Add a space between function arguments (e.g. `"message", message`). Ditto for the previous line.

Reply:
Change in next pr7